extensions/amp-story/amp-story-ads.md

```diff
@@ -0,0 +1,176 @@
+* An element ID is required, so that the template can be referenced by the ad response.
+* The selected template ID will be set as an attribute of `amp-ad`: `<amp-ad template="template-1">`
+* The content inside a template should strictly follow the [rules](https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/validator-amp-story.protoascii) of `amp-story-grid-layer`
+* Be aware of the [restrictions](../amp-mustache/amp-mustache.md#restrictions) of `amp-mustache`.
+* Ads that use different templates can be styled separately using CSS attribute selector:
+```css
+amp-ad[template=template-1] {
+  background-color: blue;
+}
+amp-ad[template=template-2] {
+  background-color: red;
+}
+```
+- The CTA (call-to-action) button should NOT
+be included in the template. Story defines a list of CTA buttons to select from.
+For details, read the ["CTA ad" section](#cta-ad) below. 
+
+### Ad response
+
+#### Response payload
+A server endpoint needs to provide ad responses in the following JSON format:
+
+```js
+{
+  "templateId": "template-1",
+  "data": {
+    "imgSrc": "https://cdn.adserver.com/img-12345.jpg",
+    "impressionUrl": "https://adserver.com/track?iid=18745543"
+  },
+  "var": {
+    "ctaType": "EXPLORE",
+    "ctaUrl": "https://advertiser.com/landing-123.html",
+    "impressionId": "ac2d1s2E3B"
+  }
+}
+```
+
+* `templateId`: the ID of the inlined template that is going to be used.
+* `data`: the data model to populate the selected template. The fields should match the variable names in the selected template.
+* `var`: extra variables needed by the story. They will be added to the `amp-ad` element as data attributes, and picked by runtime for different use cases:
+       * CTA button rendering (see details in the ["CTA ad" section](#cta-ad))
+       * Provide dynamic content of the ad for tracking purpose (see details in the "tracking" section)
+
+#### Response headers
+The ad request will be an AMPCORS request, hence a couple of custom response headers are needed.
+See [AMPCORS spec](../../spec/amp-cors-requests.md).
+
+### Tagging
+
+In AMP Story, you cannot put `amp-ad` directly onto the page, instead, all ads
+are fetched & displayed by the [amp-story-auto-ads](./amp-story-auto-ads.md)
+extension.
+
+Here is a full example using `amp-story-auto-ads` together with some templates inlined.
+
+```html
+<amp-story>
+  <amp-story-auto-ads>
+     <script type=”application/json”>
+       {
+          "ad-attributes": {
+            type: “custom”
+            data-src: “https://adserver.com/getad?slot=abcd1234”
+          }
+       }
+     </script>
+
+     <template type="amp-mustache" id="template-1">
+       <amp-img src="{{imgSrc}}"></amp-img>
+       <amp-pixel src="{{impressionUrl}}"></amp-pixel>
+     </template>
+
+     <template type="amp-mustache" id="template-2">
+       <div class="creative-line-1">{{creativeLine1}}</div>
```

Comment:
Maybe the same height width problem on `divs` not sure @lannka do you know if this will throw?

Reply:
AMP layout only applies to AMP elements

Reply:
Cool. To clarify, @jasti  no change is needed here